File: src/main/java/org/idempiere/common/base/IServiceHolder.java
package org.idempiere.common.base;

/**
 * 
 * @author hengsin
 *
 * @param <T>
 */
public interface IServiceHolder<T> {

	/**
	 * 
	 * @return service instance. null if not available or no matching service found
	 */
	public T getService();
	
}

File: src/main/kotlin/org/idempiere/common/base/IServiceHolder.kt
package org.idempiere.common.base

/**
 *
 * @author hengsin
 *
 * @param <T>
</T> */
interface IServiceHolder<T> {

    /**
     *
     * @return service instance. null if not available or no matching service found
     */
    val service: T?
}